lib/aws/active_job/sqs/poller.rb

```diff
@@ -54,30 +53,77 @@ def shutdown(timeout)
 
         def poll
           config = Aws::ActiveJob::SQS.config
-          queue = @options[:queue]
+          if @queues && !@queues.empty?
+            if @queues.size == 1
+              # single queue, use main thread
```

Comment:
Why the distinction? Could always use a background thread.

Reply:
Less resource usage - you shouldn't spawn an additional thread that we don't need.  And I want to keep encouraging this as the preferred approach.